src/globus_sdk/scopes.py

```diff
@@ -116,29 +121,52 @@ def __init__(
         self,
         resource_server: str,
         *,
-        known_scopes: t.Union[t.List[str], str, None] = None,
-        known_url_scopes: t.Union[t.List[str], str, None] = None,
+        known_scopes: ScopeBuilderScopes = None,
+        known_url_scopes: ScopeBuilderScopes = None,
     ) -> None:
         self.resource_server = resource_server
-        self._known_scopes = (
-            list(utils.safe_strseq_iter(known_scopes))
-            if known_scopes is not None
-            else []
-        )
-        self._known_url_scopes = (
-            list(utils.safe_strseq_iter(known_url_scopes))
-            if known_url_scopes is not None
-            else []
-        )
-        self._known_scope_names: t.List[str] = []
-        if self._known_scopes:
-            for scope_name in self._known_scopes:
-                self._known_scope_names.append(scope_name)
-                setattr(self, scope_name, self.urn_scope_string(scope_name))
-        if self._known_url_scopes:
-            for scope_name in self._known_url_scopes:
-                self._known_scope_names.append(scope_name)
-                setattr(self, scope_name, self.url_scope_string(scope_name))
+
+        self._registered_scope_names: t.List[str] = []
+        self._register_scopes(known_scopes, self.urn_scope_string)
+        self._register_scopes(known_url_scopes, self.url_scope_string)
+
+    def _register_scopes(
+        self, scopes: ScopeBuilderScopes, transform_func: t.Callable[[str], str]
+    ) -> None:
+        scopes_dict = self._scopes_input_to_dict(scopes)
+        for scope_name, scope_val in scopes_dict.items():
+            self._registered_scope_names.append(scope_name)
+            setattr(self, scope_name, transform_func(scope_val))
+
+    def _scopes_input_to_dict(self, items: ScopeBuilderScopes) -> t.Dict[str, str]:
+        """
+        ScopeBuilders accepts many collection-style types of scopes. This function
+          normalizes all of those types into a standard {scope_name: scope_val} dict
+
+        Translation Map:
+        None => {}
+        "my-str" => {"my-str": "my-str"}
+        ["my-list"] => {"my-list": "my-list"}
+        ("my-tuple-key": "my-tuple-val") => {"my-tuple-key": "my-tuple-val"}
```

Comment:
```suggestion
        ("my-tuple-key", "my-tuple-val") => {"my-tuple-key": "my-tuple-val"}
```

I think this is a (minor) typo.

Reply:
This line of the comment will be the end of me